config/src/main/java/org/springframework/security/config/annotation/method/configuration/MethodSecurityConfiguration.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.config.annotation.method.configuration;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ImportAware;
+import org.springframework.context.annotation.Role;
+import org.springframework.core.type.AnnotationMetadata;
+import org.springframework.security.access.annotation.SecuredAnnotationMethodSecurityAuthorizationManagerResolver;
+import org.springframework.security.access.intercept.aopalliance.MethodSecurityAuthorizationManagerAdvisor;
+import org.springframework.security.access.intercept.aopalliance.MethodSecurityAuthorizationManagerInterceptor;
+import org.springframework.security.access.method.DelegatingMethodSecurityAuthorizationManagerResolver;
+import org.springframework.security.access.method.MethodSecurityAuthorizationManager;
+import org.springframework.security.access.method.MethodSecurityAuthorizationManagerResolver;
+
+/**
+ * Base {@link Configuration} for enabling method security.
+ *
+ * @author Evgeniy Cheban
+ * @see EnableMethodSecurity
+ */
+@Configuration(proxyBeanMethods = false)
+class MethodSecurityConfiguration implements ImportAware {
+
+	private int advisorOrder;
+
+	@Bean
+	@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+	MethodSecurityAuthorizationManagerAdvisor methodSecurityAdvisor(
+			MethodSecurityAuthorizationManagerInterceptor interceptor,
+			MethodSecurityAuthorizationManagerResolver resolver) {
+		MethodSecurityAuthorizationManagerAdvisor advisor = new MethodSecurityAuthorizationManagerAdvisor(interceptor,
+				resolver);
+		advisor.setOrder(this.advisorOrder);
+		return advisor;
+	}
+
+	@Bean
+	@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+	MethodSecurityAuthorizationManagerInterceptor methodSecurityInterceptor(
+			MethodSecurityAuthorizationManager authorizationManager) {
+		return new MethodSecurityAuthorizationManagerInterceptor(authorizationManager);
+	}
+
+	@Bean
+	MethodSecurityAuthorizationManager methodSecurityAuthorizationManager(
+			MethodSecurityAuthorizationManagerResolver resolver) {
+		return new MethodSecurityAuthorizationManager(resolver);
+	}
+
+	@Bean
```

Comment:
It seems like this should be an infrastructural bean as well. Applications that need to invoke method authorization programmatically should be calling the authorization manager.

Reply:
Done.